pointcloud/basic_octree.go

```diff
@@ -129,3 +131,55 @@ func (octree *BasicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Ve
 func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
+
+// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
+func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, threshold int, buffer float64) (bool, error) {
+	if octree.MaxVal() < threshold {
+		return false, nil
+	}
+	ocbox, err := spatialmath.NewBox(
```

Comment:
[q] just confirming that NewBox takes in a center point and the dimensions of the box around it (later dividing the dimensions by 2 to give dimensions in relation to center 

(i.e. `minX = center.X - dim/2` where dim = sideLength + buffer)

Reply:
Correct